packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -19,13 +19,17 @@ import { EventEmitter } from 'events';
 
 type Func<T> = (...args: unknown[]) => T;
 
+// `any` here is intentional for variance, `unknown` won't work
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AnyFunc = (this: any, ...args: any[]) => any;
+
 /**
  * Store a map for each event of all original listeners and their "patched"
  * version. So when a listener is removed by the user, the corresponding
  * patched function will be also removed.
  */
 interface PatchMap {
-  [name: string]: WeakMap<Func<void>, Func<void>>;
+  [name: string | symbol]: WeakMap<Func<void>, Func<void>>;
```

Comment:
With better type checking after the refactors, TypeScript caught this as Node EventEmitter event types can be symbols (at least according to the types)